File: sakila-payment-service/src/main/java/com/example/sakila/module/customer/repository/CustomerRepository.java
package com.example.sakila.module.customer.repository;

import com.example.sakila.module.customer.Customer;

import java.util.List;

public interface CustomerRepository {

  Customer getCustomerById(Long id);

  List<Customer> getCustomersByStoreId(Long id);

  List<Customer> searchCustomersByFirstName(String firstName);

  List<Customer> searchCustomersByLastName(String lastName);

  Customer insertCustomer(Customer customer);

  Customer updateCustomer(Customer customer);
}

File: sakila-payment-service/src/main/java/com/example/sakila/module/customer/repository/CustomerRepository.kt
package com.example.sakila.module.customer.repository

import com.example.sakila.module.customer.Customer

interface CustomerRepository {

  fun getCustomerById(id: Long): Customer?

  fun getCustomersByStoreId(id: Long): List<Customer>?

  fun searchCustomersByFirstName(firstName: String): List<Customer>?

  fun searchCustomersByLastName(lastName: String): List<Customer>?

  fun insertCustomer(customer: Customer): Customer?

  fun updateCustomer(customer: Customer): Customer?

  fun deleteCustomer(customer: Customer)
}
